### PR TITLE
[DOCS] Refresh "Search your data"

### LINKED ISSUE #96984: [DOCS] Refresh the "Search your data" page
"[Search your data](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-your-data.html)" is one of the most clicked links on the Elasticsearch documentation landing page. However, there are a few problems with it that make it less than perfect for new users who want to learn how to search with Elasticsearch:

- it immediately jumps into the details of using the Search API.
- it lacks an overview of all of Elasticsearch' search capabilities.
- the navigation on the left has grown organically over time and lacks structure.

**Proposal**: turn "Search your data" into a landing page that provides an introductory overview of search with Elasticsearch (with links to learn more). It should:

- tell a story about supported use cases (machine-generated data vs humanly-curated data).
- cover exact value search (term queries, wildcards, numbers, ranges, geo), full text search, [semantic search](https://github.com/elastic/elasticsearch/issues/96726), and vector search.
- introduce the search API, the new Search Applications, and maybe even the different query languages as ways of searching.

The existing content on "Search your data" would move to a new sub page called "The search API. 

The new nav structure would look like this:

image

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Restructures "Search your data" to create space for content that's being migrated from the Enterprise Search docs. This PR is the first step, as we evolve "Search your data" into the starting point for all search content.

Closes https://github.com/elastic/elasticsearch/issues/96984

### CODE REVIEW

File: docs/reference/search/search-your-data/knn-search.asciidoc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [[knn-search]]
 == k-nearest neighbor (kNN) search
 ++++
-<titleabbrev>kNN search</titleabbrev>
+<titleabbrev>kNN search<</titleabbrev>
```
Comment (chriscressman) [inline docs/reference/search/search-your-data/knn-search.asciidoc:4]:
```suggestion
<titleabbrev>kNN search</titleabbrev>
```

**Review by chriscressman — APPROVED** [review]
I reviewed the search applications pages and gave everything else a quick review. Nice work!

File: docs/reference/search/search-your-data/search-application-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+  "indices": ["my_index1", "my_index2"],
+  "template": {
+    "script": {
+      "lang": "mustache",
+      "source": """
+      {
+        "query": {
+          "multi_match": {
+            "query": "{{query_string}}",
+            "fields": [{{#text_fields}}"{{name}}^{{boost}}"{{/text_fields}}]
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/search-application-api.asciidoc:153]:
```suggestion
            "fields": [{{#text_fields}}"{{name}}^{{boost}}",{{/text_fields}}]
```

File: docs/reference/search/search-your-data/search-application-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,651 @@
...
+      "source": """
+      {
+        "query": {
+          "bool": {
+            "should": [
+              {{#text}}
+              {
+                "multi_match": {
+                  "query": "{{query_string}}",
+                  "fields": [{{#text_fields}}"{{name}}^{{boost}}"{{/text_fields}}],
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/search-application-api.asciidoc:317]:
```suggestion
                  "fields": [{{#text_fields}}"{{name}}^{{boost}}",{{/text_fields}}],
```

**Comment by abdonpijpelink** [comment]
Merging this to not block follow up work. We can continue to tweak before the 8.11 release, if necessary.

**Pull request merged**